File: src/src/main/java/root/tasks/assertions/TaskCs2801.java
package root.tasks.assertions;

public class TaskCs2801 extends TaskJava2801 {

    @Override
    protected void makeLayout() {
        appendHeader();
        appendTaskDescEscaped("Дано два числа: делимое и делитель. Найти частное и остаток от деления. Все значения округлить до целого. Вывести ответ по шаблону:");
        appendCheckSingleNonEscaped("Делимое $dividend, делитель $divider, частное $quotient, остаток $remainder");
        appendCheckValuesHeader("dividend", "divider");
        appendCheckValuesRow("1234", "11");
        appendCheckValuesRow("5", "13");
        appendCheckValuesRow("17", "17");
        appendCheckValuesRow("0", "23");
        appendCheckValuesRow("4", "0");
        appendCheckValuesFooter();
        appendTaskDescNonEscaped("В этой задаче вам нужно работать только с целой частью чисел. Цифры после запятой нужно отбросить при делении. Поэтому можно воспользоваться типом \"целое число\". Переменная для целого числа объявляется так:");
        appendCheckSingleNonEscaped(escapeHtmlSymbols(getMain(getClass().getSimpleName())));
        appendFooter();
    }

    public static void main(String[] args) {
        System.out.println(new TaskCs2801());
    }
}

File: src/cs_src/TaskCs2801.cs
using System;
namespace Example
{
    class Program
    {
        static void Main(string[] args)
        {
            int dividend, divider;//Объявление переменных типа int для целых чисел
            dividend = int.Parse(Console.ReadLine());//читаем значение от пользователя
            divider = int.Parse(Console.ReadLine());
            if (divider == 0)//если значение переменной divider строго равно 0
            {
                Console.WriteLine("Делитель не может быть равен 0");
            }
            if (divider != 0)//если значение переменной divider не равно 0
            {
                int quotient = dividend / divider;
                int remainder = dividend % divider;//Нахождение остатка от деления dividend на divider
                Console.Write("Делимое {0}, делитель {1}", dividend, divider);
                Console.WriteLine(", частное {0}, остаток {1}", quotient, remainder);
            }
        }
    }
}